src/pages/signup-affiliates/_types.ts

```diff
@@ -0,0 +1,85 @@
+import { Dispatch, ReactElement, SetStateAction } from 'react'
+
+type CountryType = { name?: string; display_name?: string; prefix?: string; symbol?: string }
+export type AffiliateAccountTypes = {
+    email: string
+    account: { type: number; plan: number }
+    address_details: {
+        country: CountryType
+        state: string
+        city: string
+        street: string
+        postal_code: string
+    }
+    phone_number: { phone: string; prefix: string }
+    personal_details: {
+        username: string
+        first_name: string
+        last_name: string
+        date_birth: string
+        website_url: string
+        social_media_url: string
+        password: string
+        company_name: string
+        company_registration_number: string
+        currency: string
+    }
+    terms_of_use: {
+        non_pep_declaration: boolean
+        tnc_accepted: boolean
+        general_terms: boolean
+        is_eu_checked: boolean
+        is_partner_checked: boolean
+    }
+}
+export type setAffiliateAccountTypes = Dispatch<SetStateAction<AffiliateAccountTypes>>
+type AffiliateAccountProps = {
+    affiliate_account?: AffiliateAccountTypes
+    setAffiliateAccount?: setAffiliateAccountTypes
+    onSubmit?: () => void
+}
+export type SignUpStatusTypes =
+    | 'Username not available'
+    | 'lost connection'
+    | 'success'
+    | 'Your website is not a valid entry'
+    | ''
+export type setSignUpStatusTypes = Dispatch<SetStateAction<SignUpStatusTypes>>
+export type SignUpStatusProps = {
+    signup_status: SignUpStatusTypes
+    setSignupStatus: setSignUpStatusTypes
+} & AffiliateAccountProps
+export type SubmitTypes = {
+    is_online: boolean
+    affiliate_account: AffiliateAccountTypes
+
+    setSignupStatus: setSignUpStatusTypes
+    // once our API register this call we can remove type below
+    affiliateSend: (data?: any) => Promise<void>
```

Comment:
just a question for my own understanding why are we using any as type here?  is it like we don’t know the type?

Reply:
Sure @mitra-deriv, any for now here because we sending a lot of data, which will be typed later as for usual API call via `useWS`
API call I using here not registrated yet, but works

Reply:
@NikitK-deriv do you think we can use an `unknown` type here instead of `any`?

Reply:
@mitra-deriv  instead of that I added hardcoded types which we passing